lib/publiccloud/provider.pm

```diff
@@ -634,14 +664,15 @@ sub terraform_destroy {
         my $offer = get_var('PUBLIC_CLOUD_AZURE_OFFER');
         my $sku = get_var('PUBLIC_CLOUD_AZURE_SKU');
         my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT');
-        $cmd .= "-var 'image_id=$image' " if ($image);
-        $cmd .= "-var 'image_uri=${image_uri}' " if ($image_uri);
-        $cmd .= "-var 'offer=$offer' " if ($offer);
-        $cmd .= "-var 'sku=$sku' " if ($sku);
-        $cmd .= "-var 'storage-account=$storage_account' " if ($storage_account);
+        $vars{image_id} = $image if ($image);
+        $vars{image_uri} = $image_uri if ($image_uri);
+        $vars{offer} = $offer if ($offer);
+        $vars{sku} = $sku if ($sku);
+        $vars{'storage-account'} = $storage_account if ($storage_account);
     }
     # Ignore lock to avoid "Error acquiring the state lock"
-    $cmd .= "-lock=false ";
+    my $suffix = '-lock=false';
```

Comment:
Can't it be grouped with the other options like `-no-color- -auto-approve` ?  This way we'd avoid having a `suffix` parameter. 

Nice work!

Reply:
We'd need a suffix anyway for the case of `terraform plan` which adds `-out myplan`. But actually, I guess even in that case it wouldn't be necessary. I'm not sure.

Reply:
Fixed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20981/commits/dec7101187b27a861586404f189438f80b75bb23.